modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -767,6 +767,25 @@ Mat getMatFromTensor(const opencv_onnx::TensorProto& tensor_proto)
             Mat(sizes, CV_32FC1, val).copyTo(blob);
         }
     }
+    else if (datatype == opencv_onnx::TensorProto_DataType_FLOAT16)
+    {
+        // FIXME, for now, we only load FP16 Tensor as FP32 Mat, full support for FP16 is required in the future.
+        static bool warningOnce = true;
+        if (warningOnce)
+        {
+            CV_LOG_WARNING(NULL, "DNN: load FP16 model as FP32 model, and it takes twice the FP16 RAM requirement.");
+            warningOnce = false;
+        }
+
+        if (!tensor_proto.float_data().empty()) {
+            const ::google::protobuf::RepeatedField<float> field = tensor_proto.float_data();
+            Mat(sizes, CV_32FC1, (void*)field.data()).copyTo(blob);
+        }
+        else {
+            char* val = const_cast<char*>(tensor_proto.raw_data().c_str());
+            Mat(sizes, CV_16FC1, val).convertTo(blob, CV_32FC1);
```

Comment:
> raw_data()

We have several reports about alignment issues from ARM platforms. E.g. here: #20725
Mat of CV_16F expects 2 bytes alignment.

Reply:
Thanks for code reviewing. I will try to test this under the ARMv7 platform.

Reply:
Can this problem be fixed in another pull request? We need this feature for GSoC projects.

Reply:
@zihaomu @alalek According my research the behavior is different for different ARM generations:
- ARM v8 supports unaligned load and `vld1_s16` does not case SIGBUG, but has some performance penalty.
- ARM v7 has different behavior that depends on SoC, kernel and compiler flags and may throw SIGBUS. The behavior is managed by `/proc/cpu/alignment`.

So, I propose to ensure element-size alignment to be in safe.

Reply:
https://stackoverflow.com/questions/16548059/how-to-trap-unaligned-memory-access

Reply:
@asmorkalov Thanks for the detailed investigation. I will try to fix this.